sdk/servicebus/azure-servicebus/azure/servicebus/_common/message.py

```diff
@@ -82,170 +92,180 @@ class Message(object):  # pylint: disable=too-many-public-methods,too-many-insta
     """
 
     def __init__(self, body, **kwargs):
-        subject = kwargs.pop('subject', None)
         # Although we might normally thread through **kwargs this causes
         # problems as MessageProperties won't absorb spurious args.
         self._encoding = kwargs.pop("encoding", 'UTF-8')
-        self.properties = uamqp.message.MessageProperties(encoding=self._encoding, subject=subject)
-        self.header = uamqp.message.MessageHeader()
-        self._annotations = {}
-        self._app_properties = {}
+        self._amqp_properties = uamqp.message.MessageProperties(encoding=self._encoding)
+        self._amqp_header = uamqp.message.MessageHeader()
 
-        self.session_id = kwargs.get("session_id", None)
         if 'message' in kwargs:
             self.message = kwargs['message']
-            self._annotations = self.message.annotations
-            self._app_properties = self.message.application_properties
-            self.properties = self.message.properties
-            self.header = self.message.header
+            self._amqp_properties = self.message.properties
+            self._amqp_header = self.message.header
         else:
             self._build_message(body)
+            self.properties = kwargs.pop("properties", None)
+            self.session_id = kwargs.pop("session_id", None)
+            self.message_id = kwargs.get("message_id", None)
+            self.content_type = kwargs.pop("content_type", None)
+            self.correlation_id = kwargs.pop("correlation_id", None)
+            self.to = kwargs.pop("to", None)
+            self.reply_to = kwargs.pop("reply_to", None)
+            self.reply_to_session_id = kwargs.pop("reply_to_session_id", None)
+            self.label = kwargs.pop("label", None)
+            scheduled_enqueue_time_utc = kwargs.pop("scheduled_enqueue_time_utc", None)
+            if scheduled_enqueue_time_utc:
```

Comment:
So in reading this I found myself asking "why don't we just set this directly = like the others"; and went to look at the property setter, to which I realized if you handed "None" to scheduled_enqueue_time_utc, it appears that it would set the annotation to None.  This may still result in proper behavior, but I honestly don't know what to expect offhand.  We should make sure this is tested, regardless, to make sure it's defensive against that scenario, and assuming it's safe in that case, we can maybe chose to simplify this?

(Similar statement for the others;  I know we have a precept to not do client-side validation, but I don't know if we should even consider None a reasonable input from our perspective, or if we should sanitize it if it leads to safer interaction; I'm largely worried in case someone has code where they want to 'unset' one of these values and hands "None")

Reply:
Very good point, the `unset` part is convincing to me.

I've updated the code and test accordingly, working as expected on my locally machine.

Reply:
all right.. I dug deep into it and found more pitfalls in the **annotation** part and finally got them resolved.

There're several findings/problems in our previous implementation:

1) out-going messages can't get what it sets in the annotation due to asymmetrical key of setter/getter.
  - We wrap bytes key into `AMQP.SymbolValue` key to create entry in the annotation dict.
  - We use bytes key to retrieve info from the annotation dict (that's what amqp library gives back).
2) every time we set key-pair in annotation, we create a new entry instead of replacing the old one, which leads to buggy behavior.
3) we can't set scheduled_enqueue_time_utc to None, it leads to different bugs on different platforms.
4) The `AMQP.SymbolValue` key is a must for `scheduled_enqueue_time_utc` if we want to schedule message, while for `partition_key` and `via_partition_key` in the annotation, either key being type of `AMQP.SymbolValue` or bytes works.

Now I've made getter/setter on the annotation property consistent (partition_key, via_partition_key, scheduled_enqueue_time_utc consistent).